sale_pricelist_auto_update/models/res_partner.py

```diff
@@ -121,6 +126,12 @@ def reset_partner_pricelist(self, date_range, partner_id=False):
             """
         self._cr.execute(sql, params)
         sales_data = self._cr.dictfetchall()
+        if not sales_data and partner_id:
+            if partner_id.commercial_partner_id:
```

Comment:
No, it shouldn't, since we have `and partner_id` before `and partner_id.commercial_partner_id`.